### PR TITLE
Customizable page length menu for list view

### PR DESCRIPTION
**Pull request by alashow**
Usage (in [CrudController#setup](https://github.com/Laravel-Backpack/CRUD/blob/0be2bae4a58ce417ddd5f8865b1b6e38ee87879d/src/app/Http/Controllers/CrudController.php#L54)):
```php
$this->crud->setPageLengthMenu([100, 200, 300]);
```
or 
```php
$this->crud->setPageLengthMenu([[100, 1000, 10000, -1], ['Hundred', 'Thousand', 'Ten thousand', 'All']]);
```
Related to #1098.

### CODE REVIEW

File: src/PanelTraits/Read.php [hunk 1]
```diff
@@ -146,6 +146,34 @@ public function getDefaultPageLength()
...
     }
 
+    /**
+     * Specify array of available page lengths on the table page (list view).
+     *
+     * @param array $array 1d array of page length values,
+     *                     or 2d array (first array: page length values, second array: page length labels)
+     *                     More at: https://datatables.net/reference/option/lengthMenu
+     */
+    public function setPageLengthMenu($array)
```
Comment (lloy0076) [inline src/PanelTraits/Read.php]: Can we **not** call this `$array`?

File: src/PanelTraits/Read.php [hunk 2]
```diff
@@ -146,6 +146,34 @@ public function getDefaultPageLength()
...
+    }
+
+    /**
+     * Get page length menu for the table page (list view).
+     *
+     * @return array
+     */
+    public function getPageLengthMenu()
+    {
+        // return the custom value for this crud panel, if set using setPageLengthMenu()
```
Comment (lloy0076) [inline src/PanelTraits/Read.php]: Unnecessary comment.

File: src/PanelTraits/Read.php [hunk 3]
```diff
@@ -146,6 +146,34 @@ public function getDefaultPageLength()
...
+     *                     or 2d array (first array: page length values, second array: page length labels)
+     *                     More at: https://datatables.net/reference/option/lengthMenu
+     */
+    public function setPageLengthMenu($array)
+    {
+        $this->page_length_menu = $array;
+    }
+
+    /**
+     * Get page length menu for the table page (list view).
```
Comment (lloy0076) [inline src/PanelTraits/Read.php]: s/table page (list view)/list view/

**Comment by alashow** [comment]
@lloy0076 done.

**Pull request merged**